docs/content/concepts/assets/software-defined-assets.mdx

```diff
@@ -0,0 +1,341 @@
+def downstream_asset(upstream_asset):
+    return upstream_asset + [4]
+
+
+asset_group = AssetGroup([upstream_asset, downstream_asset])
+```
+
+To save you from needing to add each asset individually to the group, Dagster provides the <PyObject object="AssetGroup" method="from_package_name" /> method for building <PyObject object="AssetGroup" />s out of all the assets within all Python modules underneath a Python package:
+
+```python file=/concepts/assets/from_package_name.py
+from dagster import AssetGroup
+
+asset_group = AssetGroup.from_package_name("docs_snippets.concepts.assets.package_with_assets")
+```
+
+Once assets are bundled into an asset group, you can:
+
+- View them in Dagit.
+- Materialize an ad-hoc set of them in Dagit, or from Python.
+- Build a job, which materializes a fixed selections of the assets in the group, and can be put on a schedule or sensor.
+
+## Viewing and materializing assets in Dagit
+
+### Loading assets into Dagit
+
+To view and materialize assets in Dagit, you can point it at a module that contains an <PyObject object="AssetGroup" />. E.g.
+
+    dagit -m module_with_asset_group
+
+If you want Dagit to contain both an asset group and a set of [jobs](/concepts/ops-jobs-graphs/jobs-graphs) that target the assets, you can place the <PyObject object="AssetGroup" /> and jobs together inside a [repository](/concepts/repositories-workspaces/repositories).
+
+### Viewing assets in Dagit
+
+Clicking on "Assets", in the right section of Dagit's top navigation pane, takes you to the Asset Catalog, which shows a list of all your assets.
+
+<img
+alt="Asset Catalog"
+src="/images/concepts/assets/software-defined-assets/catalog.png"
+/>
+
+Clicking on the name of one of these assets will take you to the Asset Details Page for that asset.
+
+<img
+alt="Asset Details"
+src="/images/concepts/assets/software-defined-assets/details.png"
+/>
+
+You can view a graph of all the assets with their dependencies by clicking the graph icon to the upper-left of the Asset Catalog, or by clicking "View in Graph" on any of the assets.
+
+<img
+alt="Asset Graph"
+src="/images/concepts/assets/software-defined-assets/graph.png"
+/>
+
+### Materializing assets in Dagit
+
+There are a couple ways in Dagit to launch a run that materializes assets:
+
+- Navigate to the Asset Details Page for the asset and click the "Materialize" button in the upper right corner.
+- In the graph view of the Asset Catalog page, click the "Materialize" button in the upper right corner. You can click on assets to collect a subset to materialize.
+
+## Source assets
+
+It's common for software-defined assets to depend on assets that are generated somewhere else. For example, your data warehouse might contain a set of tables that another team is responsible for ingesting, but that your assets are derived them. To allow the assets in your <PyObject object="AssetGroup" /> to depend on them, you can include <PyObject object="SourceAsset" />s that represent them.
```

Comment:
some grammatical stuff here `set of tables that another team is responsible for producing, but your assets are derived from those tables`.